swabbie-redis/src/main/kotlin/com/netflix/spinnaker/swabbie/redis/RedisResourceTrackingRepository.kt

```diff
@@ -34,76 +35,107 @@ class RedisResourceTrackingRepository(
   private val clock: Clock
 ) : ResourceTrackingRepository {
 
+  private val SINGLE_RESOURCES_KEY = "{swabbie:resource}"
+  private val SOFT_DELETE_KEY = "{swabbie:softdeletes}"
+  private val DELETE_KEY = "{swabbie:deletes}"
+
+  private val log = LoggerFactory.getLogger(javaClass)
   private val redisClientDelegate: RedisClientDelegate = redisClientSelector.primary("default")
 
   override fun find(resourceId: String, namespace: String): MarkedResource? {
-    "$namespace:$resourceId".let { key ->
-      return redisClientDelegate.withCommandsClient<String> { client ->
+    val key = "$namespace:$resourceId"
+    return try {
+      redisClientDelegate.withCommandsClient<String> { client ->
         client.hget(SINGLE_RESOURCES_KEY, key)
-      }.let { json ->
-          objectMapper.readValue(json)
-        }
+      }.let { objectMapper.readValue(it) }
+    } catch (e: IllegalStateException) {
+      // resource does not exist
+      null
     }
   }
 
   override fun getMarkedResources(): List<MarkedResource> {
-    return doGetAll(true)
+    getAllIds(DELETE_KEY, true)
+      .let { ids ->
+        return hydrateMarkedResources(ids)
+      }
+  }
+
+  override fun getMarkedResourcesToSoftDelete(): List<MarkedResource> {
+    getAllIds(SOFT_DELETE_KEY, false)
+      .let { ids ->
+        return hydrateMarkedResources(ids)
+      }
   }
 
   override fun getMarkedResourcesToDelete(): List<MarkedResource> {
-    return doGetAll(false)
+    getAllIds(DELETE_KEY, false)
+      .let { ids ->
+        return hydrateMarkedResources(ids)
+      }
   }
 
-  private fun doGetAll(includeAll: Boolean): List<MarkedResource> =
-    ALL_RESOURCES_KEY.let { key ->
-      return redisClientDelegate.run {
-        this.withCommandsClient<Set<String>> { client ->
-          if (includeAll) {
-            client.zrangeByScore(key, "-inf", "+inf")
-          } else {
-            client.zrangeByScore(key, 0.0, clock.instant().toEpochMilli().toDouble())
-          }
-        }.let { set ->
-            if (set.isEmpty()) emptyList()
-            else this.withCommandsClient<Set<String>> { client ->
-              client.hmget(SINGLE_RESOURCES_KEY, *set.map { it }.toTypedArray()).toSet()
-            }.map { json ->
-                objectMapper.readValue<MarkedResource>(json)
-              }
-          }
+  override fun getIdsOfMarkedResourcesToSoftDelete(): Set<String> {
+    return getAllIds(SOFT_DELETE_KEY, false)
+  }
+
+  override fun getIdsOfMarkedResourcesToDelete(): Set<String> {
+    return getAllIds(DELETE_KEY, false)
+  }
+
+  private fun getAllIds(key: String, includeAll: Boolean): Set<String> {
```

Comment:
`includeAll` doesn't really do a good job of describing what the behavior is when false, especially when the method is called `getAllIds`, I'd expect it would always provide all ids, regardless of an `includeAll` being true or false.

Perhaps `getIds(key: String, includeFutureIds: Boolean)` or something to that effect?